### PR TITLE
Fix: using deprecated field `AddOnConfiguiration` causing `proxy health` not work.

### PR DESCRIPTION
**Pull request by xuezhaojun**
Fixes: https://github.com/open-cluster-management-io/clusteradm/issues/347

### CODE REVIEW

File: pkg/cmd/proxy/health/exec.go [hunk 1]
```diff
@@ -103,10 +103,8 @@ func (o *Options) run(streams genericclioptions.IOStreams) error {
 		return errors.Wrapf(err, "failed initializing proxy api client")
 	}
 
-	// TODO: fix this deprecated field AddOnConfiguration
-	// nolint:staticcheck
 	proxyConfig, err := proxyClient.ProxyV1alpha1().ManagedProxyConfigurations().
-		Get(context.TODO(), clusterAddon.Spec.AddOnConfiguration.CRName, metav1.GetOptions{})
+		Get(context.TODO(), "cluster-proxy", metav1.GetOptions{})
```
Comment (xuezhaojun) [inline pkg/cmd/proxy/health/exec.go:107]: The name of managedproxyconfiguration is always `cluster-proxy` if we deploy addon via clusteradm: https://github.com/open-cluster-management-io/cluster-proxy/blob/2a197b382114b674b76a362146a7259aed164424/charts/cluster-proxy/templates/managedproxyconfiguration.yaml#L4

Comment (qiujian16) [inline pkg/cmd/proxy/health/exec.go:107]: set it as a const

**Comment by xuezhaojun** [comment]
/assign @qiujian16 

Rebase, PTAL, thanks!

**Pull request merged**